libraries/SPI/src/utility/spi_com.c

```diff
@@ -407,24 +393,34 @@ spi_status_e spi_send(spi_t *obj, uint8_t *Data, uint16_t len, uint32_t Timeout)
   * @param  rx_buffer : data to receive
   * @param  len : length in byte of the data to send and receive
   * @param  Timeout: Timeout duration in tick
+  * @param  skipRecieve: skip recieving data after transmit or not
   * @retval status of the send operation (0) in case of error
   */
-spi_status_e spi_transfer(spi_t *obj, uint8_t *tx_buffer,
-                          uint8_t *rx_buffer, uint16_t len, uint32_t Timeout)
+spi_status_e spi_transfer(spi_t *obj, uint8_t *tx_buffer, uint8_t *rx_buffer,
+                          uint16_t len, uint32_t Timeout, bool skipReceive)
 {
   spi_status_e ret = SPI_OK;
-  HAL_StatusTypeDef hal_status;
+  uint32_t tickstart, size = len;
+  SPI_TypeDef *_SPI = obj->handle.Instance;
 
-  if ((obj == NULL) || (len == 0)) {
-    return SPI_ERROR;
+  if ((obj == NULL) || (len == 0) || (Timeout == 0U)) {
+    return Timeout > 0U ? SPI_ERROR : SPI_TIMEOUT;
   }
-
-  hal_status = HAL_SPI_TransmitReceive(&(obj->handle), tx_buffer, rx_buffer, len, Timeout);
-
-  if (hal_status == HAL_TIMEOUT) {
-    ret = SPI_TIMEOUT;
-  } else if (hal_status != HAL_OK) {
-    ret = SPI_ERROR;
+  tickstart = HAL_GetTick();
+  while (size--) {
+    while (!LL_SPI_IsActiveFlag_TXE(_SPI))
```

Comment:
This LL API is not available in H7 and MP1.

Reply:
ive checked and found *_ll_spi.* is available for H7 and MP1
but yes, i found some differences for H7 at least, will check it